src/crimson/os/seastore/extent_placement_manager.h

```diff
@@ -220,7 +230,7 @@ class ExtentPlacementManager {
 	      gen};
     };
 
-    if (!is_logical_type(type)) {
+    if (should_inline(type)) {
```

Comment:
IIUC the situation after this PR is that all extents with RBM must be OOL.

If true, probably we can adjust to the following logic with the explict restrictions:
```c++
if (type == extent_types_t::ROOT) {
  return alloc_paddr(INLINE_GENERATION, data_category_t::METADATA, length);
}

if (background_process.get_backend_type() == backend_type_t::SEGMENTED &&
    is_lba_backref_node(type)) {
  // with SEGMENTED, lba-backref extents must be INLINE
  return alloc_paddr(INLINE_GENERATION, data_category_t::METADATA, length);
}

if (hint == placement_hint_t::COLD) {
  assert(gen == INIT_GENERATION);
  return alloc_paddr(MIN_REWRITE_GENERATION, get_extent_category(type), length);
}

if (get_extent_category(type) == data_category_t::METADATA &&
    gen == INIT_GENERATION) {
  if (background_process.get_backend_type() == backend_type_t::SEGMENTED) {
    // with SEGMENTED, default not to ool metadata extents to reduce padding overhead.
    // TODO: improve padding so we can default to the ool path.
    return alloc_paddr(INLINE_GENERATION, get_extent_category(type), length);
  } else {
    // with RBM, all extents must be OOL
    assert(background_process.get_backend_type() == backend_type_t::RANDOM_BLOCK);
    return alloc_paddr(OOL_GENERATION, get_extent_category(type), length);
  }
} else {
  assert(get_extent_category(type) == data_category_t::DATA ||
         gen >= MIN_REWRITE_GENERATION);
  if (gen > MAX_REWRITE_GENERATION) {
    gen = MAX_REWRITE_GENERATION;
  } else if (gen == INIT_GENERATION) {
    gen = OOL_GENERATION;
  }
  return alloc_paddr(gen, get_extent_category(type), length);
}
```

Related adjustments:
* Drop the deprecated prefer_ool.
* Introduce BackgroundProcess::get_backend_type(), should be available from trimmer.
* Introduce is_lba_backref_node() in seastore_types.h

Reply:
cc @zhscn